tests/system/test_system_async.py

```diff
@@ -0,0 +1,1000 @@
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import datetime
+import math
+import pytest
+import operator
+import os
+import re
+
+from google.oauth2 import service_account
+
+from google.api_core.exceptions import AlreadyExists
+from google.api_core.exceptions import FailedPrecondition
+from google.api_core.exceptions import InvalidArgument
+from google.api_core.exceptions import NotFound
+from google.cloud._helpers import _datetime_to_pb_timestamp
+from google.cloud._helpers import UTC
+from google.cloud import firestore_v1 as firestore
+from test_utils.system import unique_resource_id
+
+FIRESTORE_CREDS = os.environ.get("FIRESTORE_APPLICATION_CREDENTIALS")
+FIRESTORE_PROJECT = os.environ.get("GCLOUD_PROJECT")
+RANDOM_ID_REGEX = re.compile("^[a-zA-Z0-9]{20}$")
+MISSING_DOCUMENT = "No document to update: "
+DOCUMENT_EXISTS = "Document already exists: "
+UNIQUE_RESOURCE_ID = unique_resource_id("-")
```

Comment:
Is this id unique to the scope of all tests, or unique for each test (I'm not familiar with the lifecycle of pytest files)?

I think we'll want this to be unique for each test, correct? If it's not already scoped that way we might need to move this to the fixture so that it is.

Reply:
This is something I mimicked from the sync system tests. I believe it is unique to the scope of all tests, though each test individually defines a prefix to make the id unique for each test. Is there a compelling reason to generated a unique id postfix for all tests?

Reply:
I think borrowing from sync should be fine here.

Reply:
Sounds good